src/bte.f90

```diff
@@ -1064,6 +1108,21 @@ subroutine iterate_bte_el(T, drag, num, el, rta_rates_ibz, field_term, &
                   response_el(aux, n, :)*(Xplus(iproc) + Xminus(iproc))
           end do
 
+          !Add charged impurity contribution to the self term
+          if(num%elchimp) then
+             do iproc = 1, nprocs_echimp
+                 !Grab the final electron and, if needed, the interacting phonon
+                 call demux_state(istate_el_echimp(iproc), numbands, n, ikp)
+
+                 !Self contribution:
+                 !Find image of final electron wave vector due to the current symmetry
+                 call binsearch(el%indexlist, el%equiv_map(ik_sym, ikp), aux)
+
+                 response_el_reduce(ik_fbz, m, :) = response_el_reduce(ik_fbz, m, :) + &
+                      response_el(aux, n, :) * Xchimp(iproc)
+             end do
+          end if
+
```

Comment:
Regarding point 2, I've checked and in principle is equivalent to the isotopic case.
As for point 1, can we maybe take into account that since we are messing with that part of the code to redo it, maybe taking into consideration the implementation of EPW for impurity scattering (see `rgd_imp_epw_fine` within `rigid_epw.f90`, but summing over G I think, I'll need to check that) plus a Thomas-Fermi correction to remove the singularity. So, the electron wavefunctions are taken into account, plus it would be applicable to anisotropic cases. It will still have the limitation of being 3D-bounded, but 2D electrostatics would require a new full chapter. 

Or we can do it as you say, and leave it for another day.

Reply:
Yes, the electron-ch. imp. scattering calculation can be improved. You have already mentioned this in your comment. I'll add here that the expression we want to implement is Eq. 26 of Hyungjun Lee, et al. [ https://arxiv.org/pdf/2302.08085.pdf ]. There, the overlap integral is given by Eq. 13. We already have all the quantities we need in `elphbolt`. (An example of this type of overlap integral can be found in `wannier.f90`.) This, I think, will be a big improvement over the simple expression currently being used, especially for capturing any anisotropy. 

I leave it up to you to decide if you want to take this on as a task now.  I have been working on improving the scattering theory anyway, but focusing first on the phonon-impurity problem before getting to the electron-impurity.